### ISSUE TITLE
Color function and percents seems confusing

### ISSUE THREAD

**Issue by facelessuser** [LeaVerou/color.js#58]
Consider the following (outputs as comments):

```js
new Color("color(srgb 100% 50% 30%)"); // rgb(100% 50% 30%)
new Color("color(lab 100% 50 30)"); // lab(1% 50 30)
```

I think I understand the intention:

1. Percents in color functions will always be translated to a range between 0 and 1.
2. Floats/integers are always taken at face value.

So with sRGB, this maps to sane values because they store their data in the range 0 - 1, but with Lab 100% maps to 1% as that space does not scale the color channels which makes sense as, for instance, `a` and `b` are theoretically unbound so they do not reasonably fit the mold that sRGB channels do.

So here are the questions:

1. Should percent syntax **not** be used with the color function if using color spaces Lab, LCH, HSL, HWB, etc?
2. Are percents in this case supposed to work for Lab, LCH, and others, but scaled differently than RGB color spaces?
3. Should colors like Lab scale their percent channels to be between 0 - 1 internally, but not the other channels?
4. Is this a one-way conversion? So 100% will translate to 1 but if you enter `color(lab 1 0 0)` you will not get 100%, but you should get 1%? And if 1 should translate to 100%, do you **not** scale the `a` channel of 50 to be 100 * 50 (meaning that the logic differs based on the channel)?

Basically, I can't tell if the translation of percents is supposed to be just a general handling of percent inputs in color functions, or if it is supposed to be color space aware. It is unclear looking at the level 4 color spec, so I'm not sure if the implementation here represents reality or not.

**Comment by svgeesus** [comment]
CIE Lightness is defined to use a 0 to 100 scale, where 100 is the same luminance as the reference white.
Industry practice is to not use this as a type of chromatic adapatation, but to first use a Bradford transform to transform to D50; and the reference white is D50 white.

> with Lab 100% maps to 1%

This is why, when I finally caved on adding % to CIE Lightness  in CSS Color 4 (which is more CSS-like, but goes against entire industry practice) I made it mandatory. So there is no confusion with L=1 meaning 100 or 1, because it always takes a percentage. (And it is a percentage, I guess, of the luminance level of the reference diffuse white).

color.js though allows both 0-100 and 0-100% when specifying a CIE Lightness. If someone uses percent, they are most likely thinking of the CSS syntax, so I want 100% to map to  a Lightness of 100, not 1.

If I convert, say, `papayawhip` to Lab I get coords of [95.229, 2.539, 14.676]. `color.toString()` wil output `lab(95.229% 2.539 14.676)` (but `Color.prototype.toString.call(color)` will indeed output `color(lab 95.229 2.539 14.676)` rather than the CSS-compatible `color(lab 95.229% 2.539 14.676)`).

> ```new Color("color(lab 100% 50 30)"); // lab(1% 50 30)```

That is unintended, and a bug.

**Comment by facelessuser** [comment]
Great! This clarifies things quite a bit, so there should be a little awareness for percentage only channels.

**Comment by svgeesus** [comment]
A similar comment [was made about CSS Color 4](https://github.com/w3c/csswg-drafts/issues/5894) which resulted in [explicitly stating where percentages were allowed, required or forbidden](https://github.com/w3c/csswg-drafts/commit/b522279ce3a03ca6dc1bd2a5957bc4fc49c184b5#diff-70d5313ae3d365ed6982af36c8980a1bd25b3ad96b5dd277b0aa892c357c9523)

Related, I noticed this too without remembering this issue, so [Percentages mishandled in `color(lab L a b)`](https://github.com/LeaVerou/color.js/issues/70).

I also noticed a similar error in a CSS implementation, which was treating 50% as L=0.5 not L=100.

_My preference would have been to not allow percent, which is merely a source of confusion and non-interoperability._

**Comment by facelessuser** [comment]
Thanks for the heads up. The spec being explicit about this kind of thing really takes out the guess work, so much appreciated!

**Comment by svgeesus** [comment]
CSS Color 4 now defines explicit reference ranges for each color space, so that interconversion between numbers and percents is clearly defined in all cases. For example [for OKLab and OKLCH](https://drafts.csswg.org/css-color-4/#specifying-oklab-oklch)

**Comment by facelessuser** [comment]
Yep, I think I need to clean up and commit those new CSS changes. I think things are pretty clear now.
